#VRML_SIM R2020a utf8
WorldInfo {
  ERP 0.15
  physics "collision_detector"
  basicTimeStep 10
  FPS 20
  defaultDamping Damping {
  }
  #coordinateSystem "NUE"
  lineScale 0.01
  contactProperties [
    DEF floor-ball ContactProperties {
      material1 "floor"
      material2 "ball"
      coulombFriction [
        0.1
      ]
      bounce 0.3
      softERP 0.15
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF floor-body ContactProperties {
      material1 "floor"
      material2 "body"
      coulombFriction [
        0
      ]
      bounce 0
      softERP 0.15
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF floor-head ContactProperties {
      material1 "floor"
      material2 "head"
      coulombFriction [
        0
      ]
      bounce 0
      softERP 0.15
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF floor-wheel ContactProperties {
      material1 "floor"
      material2 "wheel"
      coulombFriction [
        3
      ]
      bounce 0
      softERP 0.15
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-ball ContactProperties {
      material1 "wall"
      material2 "ball"
      bounce 0.3
      softERP 1
      softCFM 0
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-body ContactProperties {
      material1 "wall"
      material2 "body"
      bounce 0.1
      softERP 0.025
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-head ContactProperties {
      material1 "wall"
      material2 "head"
      bounce 0.1
      softERP 0.025
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wall-wheel ContactProperties {
      material1 "wall"
      material2 "wheel"
      bounce 0.1
      softERP 0
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF goal-ball ContactProperties {
      material1 "goal"
      material2 "ball"
      bounce 0.05
      softERP 0
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF goal-body ContactProperties {
      material1 "goal"
      material2 "body"
      bounce 0.1
      softERP 0.025
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF goal-head ContactProperties {
      material1 "goal"
      material2 "head"
      bounce 0.1
      softERP 0.025
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF goal-wheel ContactProperties {
      material1 "goal"
      material2 "wheel"
      bounce 0.1
      softERP 0
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF ball-body ContactProperties {
      material1 "ball"
      material2 "body"
      bounce 0
      softERP 0.8
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF ball-head ContactProperties {
      material1 "ball"
      material2 "head"
      bounce 1
      softERP 0.8
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF ball-wheel ContactProperties {
      material1 "ball"
      material2 "wheel"
      bounce 0
      softERP 0
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF body-body ContactProperties {
      material1 "body"
      material2 "body"
      bounce 0.25
      softERP 0.025
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF body-head ContactProperties {
      material1 "body"
      material2 "head"
      bounce 0.25
      softERP 0.025
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF body-wheel ContactProperties {
      material1 "body"
      material2 "wheel"
      bounce 0
      softERP 0.3
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF head-head ContactProperties {
      material1 "head"
      material2 "head"
      bounce 0.25
      softERP 0.025
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF head-wheel ContactProperties {
      material1 "head"
      material2 "wheel"
      bounce 0
      softERP 0.3
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    DEF wheel-wheel ContactProperties {
      material1 "wheel"
      material2 "wheel"
      bounce 0
      softERP 0
      softCFM 1e-03
      bumpSound ""
      rollSound ""
      slideSound ""
    }
  ]
}
DEF DEF_AUDVIEW Viewpoint {
  description "2"
  orientation -1 0 0 0.8910057735495215
  position 0 11 9
  follow ""
  followSmoothness 0.05
}
DirectionalLight {
  ambientIntensity 0.7
  color 0.845624 0.881132 0.976471
  direction 2 -7 -2
  intensity 2
  castShadows TRUE
}
DirectionalLight {
  ambientIntensity 1
  color 0.5 0.5 0.5
  direction -1 0 -1
  castShadows TRUE
}
DirectionalLight {
  ambientIntensity 1
  color 0.5 0.5 0.5
  direction 1 0 1
  castShadows TRUE
}
DEF DEF_BALLSPOTLIGHT Solid{
  translation 0 0.95 0
  name "spotlight"
  children[
    DEF DEF_SPOTLIGHT SpotLight{
      attenuation 0 0 0.15
      beamWidth 0.6
      color 0 1 0
      cutOffAngle 0.6
      direction 0 -1 0
      intensity 1
      location 0 0 0
      radius 1.15
    }
  ]
}
Referee {
  translation 0 25.6 0
  children [
    DEF DEF_RECV Receiver {
      name "recv"
    }
    Transform {
      rotation 1 0 0 4.712389
      children [
        DEF DEF_CAMA Camera {
          rotation 0 0 1 0
          name "cam_a"
          fieldOfView 0.43
          width 640
          height 380
          near 20
        }
        DEF DEF_CAMB Camera {
          rotation 0 0 1 3.1415926
          name "cam_b"
          fieldOfView 0.43
          width 640
          height 380
          near 20
        }
      ]
    }
  ]
  controller "supervisor"
}
SoccerField5v5 {
  fieldLength 10.14
  fieldWidth 6.046
  floorLength 11.44
  floorWidth 6.24
  centerDotRadius 0.013
  centerCircleRadius 0.65
  penaltyAreaDepth 1.17
  penaltyAreaWidth 2.34
  goalAreaDepth 0.52
  goalAreaWidth 1.69
  lineThickness 0.04
  soccerFieldWall [
    Group {
      children [
        DEF DEF_WALL SoccerFieldWall {
          length 10.14
          width 6.046
          cornerLength 0.13
          goalHight 0.52
          goalWidth 1.3
          goalDepth 0.585
          wallHeight 0.075
          wallAppearance Gray {
          }
          wallContactMaterial "wall"
          goalContactMaterial "goal"
        }
        DEF DEF_VISWALL VisualWall {
          length 10.14
          width 6.045
          goalWidth 1.3
          gpTranslation 5.3053
          gpScale 2.8509
          bannerBehind TRUE
          bannerHorizontal 10
          cornerLength 0.13
        }
      ]
    }
  ]
  grassSlot DEF DEF_GRASS SoccerFieldGrass {
    blockSize 0.65
    length 11.7
    width 7.8
  }
  floorContactMaterial "floor"
}
DEF DEF_BALL SoccerBall {
  translation 0 0.06 0
  contactMaterial "ball"
  shapes [
    DEF DEF_BALLSHAPE SoccerBallShape {
      radius 0.05
    }
    DEF DEF_ORANGESHAPE SoccerBallOrangeShape {
      radius 0.05
    }
  ]
}

DEF DEF_ROBOTR0 SoccerRobot {
  translation -2.925 0.045 -1.3
  rotation 0 1 0 -1.5708
  name "R1"
  customData "0 0 0 0 0 0 0 0"
  controller "soccer_robot"
  #slipNoise 0
  bodyAppearance RedPaint{ }
  metalAppearance RedMetal{ }
  tireAppearance Gray { }
  wheelAppearance RedPaint{ }
  bodyContactMaterial "body"
  headContactMaterial "head"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      role "D1"
    }
  ]
}
DEF DEF_ROBOTB0 SoccerRobot {
  translation 2.925 0.045 1.3
  rotation 0 1 0 1.5708
  name "B1"
  customData "0 0 0 0 0 0 0 0"
  controller "soccer_robot"
  #slipNoise 0
  bodyAppearance BluePaint { }
  metalAppearance BlueMetal{ }
  tireAppearance Gray { }
  wheelAppearance BluePaint { }
  bodyContactMaterial "body"
  headContactMaterial "head"
  wheelContactMaterial "wheel"
  patches [
    SoccerRobotNumberPatch {
      role "D1"
      isTeamTagRed FALSE
    }
  ]
}